components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2011,6 +2017,43 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return ports;
     }
 
+    public async updateRepoStatus(
+        ctx: TraceContext,
+        workspaceId: string,
```

Comment:
Why not make the API based on instanceId? That would get rid of potential bad writes because of timing issues between here and bridge.
Also, having the mutex on instanceId is better as well. :+1:

Ideally, we'd even make the git.status a separate field on the DB, so we would not even need the mutex at all... :slight_smiling_face:

Reply:
If we introduce another property we should remove the git information from the status completely

Reply:
After we remove feature flag though and migrate all clients to a new property?

Reply:
@geropl @svenefftinge I don't think I would have time to apply changes. If you can pick up, or it will need to wait till JB is sorted out.